pynbody/snapshot/__init__.py

```diff
@@ -1908,13 +1908,14 @@ def __init__(self, base, index_array=None, iord_array=None):
     def _iord_to_index(self, iord):
         # Maps iord to indices. Note that this requires to perform an argsort (O(N log N) operations)
         # and a binary search (O(M log N) operations) with M = len(iord) and N = len(self.base).
-        if 'iord_argsort' not in self.base:
-            self.base['iord_argsort'] = np.argsort(self.base['iord'])
+
+        if not util.is_sorted(iord) == 1:
+            raise Exception('Expected iord to be sorted in increasing order.')
```

Comment:
As far as I can tell, it seems like it's still ok if iord isn't sorted, since `iord_argsort` takes care of it? But maybe I missed something?

Reply:
Unfortunately no, because `util.binary_search` expects its first argument to be sorted in increasing order (unlike `np.search_sorted`).

Reply:
Maybe I made a poor choice of variable name ; `iord_argsort` is an array that sorts `iord_base`, not `iord`. I am changing the variable names

Reply:
Ah I see, yes!